test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ComplexTypeInheritance/ComplexTypeInheritanceControllers.cs

```diff
@@ -117,6 +117,25 @@ public ITestActionResult Patch(int key, [FromBody]Delta<Window> delta)
             return Ok(window);
         }
 
+        [ODataRoute("Windows({key})/CurrentShape")]
+        public ITestActionResult PatchShape(int key, [FromBody] Delta<Shape> delta)
+        {
+            Window window = _windows.FirstOrDefault(e => e.Id == key);
```

Comment:
Given that there is no `null` check, `First` makes more sense than `FirstOrDefault`